inorbit_edge/robot.py

```diff
@@ -899,72 +902,106 @@ def publish_odometry(
+            raise ValueError("Input lists must be of the same length")
+        elif not self._should_publish_message(method="publish_lasers"):
             return None
 
-        pb_lasers_message = LaserMessage()
-        pb_lasers_message.name = "0"
-
-        # Encode ranges list using a compact representation
-        runs, values = encode_floating_point_list(ranges)
-
-        # Update LaserMessage message with encoded laser ranges
-        pb_lasers_message.ranges.runs.extend(runs)
-        pb_lasers_message.ranges.values.extend(values)
-
-        # Populate LocationAndPoseMessage with current pose
-        # and laser data, encoded as floating point list.
+        # Populate LocationAndPoseMessage with current pose and laser data,
+        # encoded as a floating point list.
         msg = LocationAndPoseMessage()
         msg.ts = ts if ts else int(time.time() * 1000)
         msg.pos_x = x
         msg.pos_y = y
         msg.yaw = yaw
         msg.frame_id = frame_id
-        msg.lasers.append(pb_lasers_message)
 
-        # Publish laser configuration, based on provided and/or infered parameters.
-        # Note: x, y & yaw should be used if there is a robot to laser transform.
-        #   As this is not supported they are explicitely set to zero.
-        self.publish(
-            topic=self._get_robot_subtopic(MQTT_SUBTOPIC_LASER_CONFIG),
-            message=(
-                "{ts:d}|{x:.4g}|{y:.4g}|{yaw:.6g}|{angle_min:.6g}|{angle_max:.6g}|"
-                "{range_min:.4g}|{range_max:.4g}|{n_points:d}"
-            ).format(
-                ts=int(time.time() * 1000),
-                x=0,
-                y=0,
-                yaw=0,
-                angle_min=angle[0],
-                angle_max=angle[1],
-                range_min=min(pb_lasers_message.ranges.values),
-                range_max=max(pb_lasers_message.ranges.values),
-                n_points=len(ranges),
-            ),
-            qos=1,
-            retain=True,
-        )
+        # Go through each of the laser streams
+        for i, current in enumerate(ranges):
+            # Names are based on the index
+            name = str(i)
+
+            pb_lasers_message = LaserMessage()
+            pb_lasers_message.name = name
+
+            # Encode ranges list using a compact representation
+            runs, values = encode_floating_point_list(current)
+
+            # Update LaserMessage message with encoded laser ranges
+            pb_lasers_message.ranges.runs.extend(runs)
+            pb_lasers_message.ranges.values.extend(values)
+
+            msg.lasers.append(pb_lasers_message)
+
+            # Publish laser configuration once.
+            # Based on provided and/or inferred parameters.
+            # Note: x, y & yaw should be used if there is a robot to laser transform.
+            #   As this is not supported they are explicitly set to zero.
+            topic = MQTT_SUBTOPIC_LASER_CONFIG_BASE + name
```

Comment:
I'd move this to a separate method to make this one easier to read e.g. `_publish_laser_config_once`